src/test/java/com/exasol/swqa/maven/SummarizerMojoIT.java

```diff
@@ -0,0 +1,134 @@
+import static java.time.temporal.ChronoUnit.MILLIS;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+
+import org.apache.maven.it.Verifier;
+import org.junit.jupiter.api.*;
+import com.exasol.mavenpluginintegrationtesting.MavenIntegrationTestEnvironment;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.Duration;
+import java.util.Comparator;
+
+// [itest -> dsn~os-compatibility~1]]
+class SummarizerMojoIT {
+    private static final String GROUP_ID = "com.exasol";
+    private static final String ARTIFACT_ID = "quality-summarizer-maven-plugin";
+    private static final String GOAL = "summarize";
+    private static final String PLUGIN_VERSION = getCurrentProjectVersion();
+    private static final String PLUGIN_JAR_FILE_NAME = ARTIFACT_ID + "-" + PLUGIN_VERSION + ".jar";
+    private static final String MAVEN_GOAL = GROUP_ID + ":" + ARTIFACT_ID + ":" + PLUGIN_VERSION + ":" + GOAL;
+    private static final Path PLUGIN_JAR = Path.of("target", PLUGIN_JAR_FILE_NAME).toAbsolutePath();
+    // We need the flattened version of the POM file in order to include the generated POM.
+    private static final Path PLUGIN_POM = Path.of(".flattened-pom.xml").toAbsolutePath();
+    private static final Path BASE_TEST_DIR = Paths.get("src/test/resources").toAbsolutePath();
+
+    private static MavenIntegrationTestEnvironment testEnvironment;
+
+    @BeforeAll
+    // [itest -> dsn~maven-plugin~1]
+    static void beforeAll() {
+        testEnvironment = new MavenIntegrationTestEnvironment();
+        testEnvironment.installPlugin(PLUGIN_JAR.toFile(), PLUGIN_POM.toFile());
+    }
+
+    @CsvSource({ "10, 0, 0.0", "20, 80, 80.0", "20, 40, 66.7", "0, 100, 100.0", })
+    @ParameterizedTest
+    // [itest -> dsn~extracting-code-coverage-from-ja-co-co-report~1]
+    // [itest -> dsn~metric-output-file~1]
+    // [itest -> dsn~writing-code-coverage-value~1]
+    void testExtractPathCoverage(final int missed, final int covered, final float expectedPercentage) throws Exception {
+        final Path baseTestDir = BASE_TEST_DIR.resolve("project-with-coverage");
+        final Path targetDir = baseTestDir.resolve("target");
+        final Path siteDir = prepareSiteDir(targetDir);
+        createMetricsFile(siteDir, missed, covered);
+        // [itest -> qs~allowed-execution-time~1]
+        runMojoWithTimeout(baseTestDir, Duration.of(500, MILLIS));
+        assertThat(Files.readString(targetDir.resolve("metrics.json")), equalTo("""
+                {
+                    "coverage" : %.1f
+                }
+                """.formatted(expectedPercentage)));
+        cleanUpSiteDir(siteDir);
```

Comment:
Either
* copy complete project to a `@TempDir` that is cleaned up automatically
* or put `cleanUpSiteDir()` into
  * a `finally` block
  * `@AfterEach`